src/com/SocketSendQueue.cpp

```diff
@@ -23,24 +23,32 @@ void SocketSendQueue::dispatch(std::shared_ptr<Socket>      sock,
                                boost::asio::const_buffers_1 data,
                                std::function<void()>        callback)
 {
+  std::lock_guard<std::mutex> lock(_queueMutex);
```

Comment:
Stupid question: Couldn't this lead to a deadlock when locking the mutex here and trying to lock it later down in the `process()->sendComplete()` function? We had an extra scope for the lock previously.

Reply:
That's not a stupid question. Boost.asio guarantees that the handler (or the lambda in this case) will not be invoked from the calling function.

> Regardless of whether the asynchronous operation completes immediately or not, the handler will not be invoked from within this function. On immediate completion, invocation of the handler will be performed in a manner equivalent to using [post](https://www.boost.org/doc/libs/1_71_0/doc/html/boost_asio/reference/post.html).

[Source](https://www.boost.org/doc/libs/1_71_0/doc/html/boost_asio/reference/async_write/overload1.html)

Reply:
Alright, thanks